src/components/List/ListPanel/ListPanel.spec.js

```diff
@@ -0,0 +1,178 @@
+import { defaultQuery } from '../../../reducers/query/query';
+import { defaultResults } from '../../../reducers/results/results';
+import { defaultView } from '../../../reducers/view/view';
+import * as utils from '../../../utils';
+import * as pagingActions from '../../../actions/paging';
+
+describe('ListPanel', () => {
+  const renderComponent = (
+    newAggsState,
+    newQueryState,
+    newResultsState,
+    newViewState
+  ) => {
+    merge(newAggsState, defaultAggs);
+    merge(newQueryState, defaultQuery);
+    merge(newResultsState, defaultResults);
+    merge(newViewState, defaultView);
+    const data = {
+      aggs: newAggsState,
+      query: newQueryState,
+      results: newResultsState,
+      view: newViewState,
+    };
+
+    render(
+      <IntlProvider locale="en">
+        <ListPanel />
+      </IntlProvider>,
+      {
+        preloadedState: data,
+      }
+    );
+  };
+
+  const analyticsSpy = jest
+    .spyOn(utils, 'sendAnalyticsEvent')
+    .mockImplementation(() => jest.fn());
+  const itemFixture = {
+    product:
+      'Credit reporting, credit repair services, or other personal consumer reports',
+    complaint_what_happened: '',
+    date_sent_to_company: '2022-11-16T12:00:00-05:00',
+    issue: 'Incorrect information on your report',
+    sub_product: 'Credit reporting',
+    zip_code: '12345',
+    tags: null,
+    has_narrative: false,
+    complaint_id: '7990095',
+    timely: 'Yes',
+    consumer_consent_provided: null,
+    company_response: 'In progress',
+    submitted_via: 'Web',
+    company: 'JP Morgan',
+    date_received: '2022-11-16T12:00:00-05:00',
+    state: 'FL',
+    consumer_disputed: 'N/A',
+    company_public_response: null,
+    sub_issue: 'Public record information inaccurate',
+  };
+
+  test('Render ListPanel with no results', () => {
+    const newAggsState = {
+      error: '',
+    };
+    const newResultsState = {
+      isLoading: false,
+      items: [],
+    };
+
+    renderComponent(newAggsState, defaultQuery, newResultsState, defaultView);
+
+    expect(
+      screen.getByRole('heading', {
+        name: /No results were found for your search/,
+      })
+    ).toBeDefined();
+  });
+
+  test('Render ListPanel with an error', () => {
+    const newAggsState = {
+      error: { message: 'error message', name: 'messageTypeName' },
+    };
+
+    renderComponent(newAggsState, defaultQuery, defaultResults, defaultView);
+
+    expect(
+      screen.getByText(/There was a problem executing your search/)
+    ).toBeDefined();
+  });
+
+  test('Render ListPanel with an item', () => {
+    const newAggsState = {
+      error: '',
+    };
+    const newResultsState = { items: [itemFixture] };
+
+    renderComponent(newAggsState, defaultQuery, newResultsState, defaultView);
+
+    expect(screen.getByText('JP Morgan')).toBeDefined();
+    expect(screen.getByText('11/16/2022')).toBeDefined();
+  });
+
+  test('onSize triggers dispatch and analtyics event', () => {
```

Comment:
this is so much better and more thorough than the previous method of testing mapDispatchtoProps

🍾